src/Tooltip/Tooltip.tsx

```diff
@@ -1,81 +1,113 @@
 import "./Tooltip.scss";
 
 interface IProps {
 	title: string | JSX.Element;
 	position?: "top" | "bottom";
+	showByClick?: boolean;
 	className?: string;
 }
 
-interface IState {
-	isShow: boolean;
-}
-
-export class Tooltip extends React.Component<IProps, IState> {
-	public state = {
-		isShow: false
-	};
+const WithOutsideClickOverflowVisibleContainer = withOutsideClick(
+	OverflowVisibleContainer
+);
 
-	public refTitle = React.createRef<HTMLDivElement>();
+export const Tooltip: React.FC<IProps> = ({
+	title,
+	position = "bottom",
+	className,
+	children,
+	showByClick = false
+}) => {
+	const [isShow, setIsShow] = React.useState(false);
+	const refTitle = React.useRef<HTMLDivElement>(null);
 
-	public handleShowTooltip = () => {
-		this.setState({ isShow: true });
+	const handleShowTooltip = () => {
+		setIsShow(true);
 	};
-	public handleHideTooltip = () => {
-		this.setState({ isShow: false });
-	};
-
-	public render() {
-		const { isShow } = this.state;
-		const { title, position = "bottom", className, children } = this.props;
 
-		if (!children) {
-			return (
-				<div className={cn("kit-tooltip", className)}>
-					<div className="kit-tooltip__title">{title}</div>
-				</div>
-			);
-		}
+	const handleHideTooltip = () => {
+		setIsShow(false);
+	};
 
+	if (!children) {
 		return (
 			<div className={cn("kit-tooltip", className)}>
-				<div
-					onMouseEnter={this.handleShowTooltip}
-					onMouseLeave={this.handleHideTooltip}
-					ref={this.refTitle}
-					className="kit-tooltip__title"
-				>
-					{isShow && (
-						<span
-							className={cn(
-								"kit-tooltip__road",
-								`kit-tooltip__road_${position}`
-							)}
-						/>
-					)}
-					{title}
-				</div>
-				<OverflowVisibleContainer
-					parentRef={this.refTitle}
-					className="kit-tooltip__popup"
+				<div className="kit-tooltip__title">{title}</div>
+			</div>
+		);
+	}
+
+	const tooltipContent = (
+		<div
+			onMouseEnter={showByClick ? undefined : handleShowTooltip}
+			onMouseLeave={showByClick ? undefined : handleHideTooltip}
+			className={cn(
+				"kit-tooltip__content",
+				`kit-tooltip__content_${position}`,
+				{
+					"kit-tooltip__content_show": isShow
+				}
+			)}
+		>
+			{showByClick && (
+				<button
+					type="button"
+					onClick={handleHideTooltip}
+					className="kit-tooltip__close"
 				>
-					<div
-						onMouseEnter={this.handleShowTooltip}
-						onMouseLeave={this.handleHideTooltip}
+					<IconSvg type="close" className="kit-tooltip__close-icon" />
+				</button>
+			)}
+			{children}
+		</div>
+	);
+
+	return (
+		<div className={cn("kit-tooltip", className)}>
+			<div
+				onMouseEnter={showByClick ? undefined : handleShowTooltip}
+				onMouseLeave={showByClick ? undefined : handleHideTooltip}
+				onClick={showByClick ? handleShowTooltip : undefined}
+				ref={refTitle}
+				className={cn("kit-tooltip__title", {
+					"kit-tooltip__title_pointer": showByClick
+				})}
+			>
+				{isShow && (
```

Comment:
Мостик этот не нужен при открытии по клику
```suggestion
				{isShow && !showByClick && (
```